Remove JodaCompatibleZonedDateTime (#78417)
elastic/elasticsearch: docs/reference/migration/migrate_8_0/scripting.asciidoc

[discrete]
[[breaking_80_scripting_changes]]
==== Scripting changes

//NOTE: The notable-breaking-changes tagged regions are re-used in the
//Installation and Upgrade Guide

//tag::notable-breaking-changes[]

> .JodaCompatibleDateTime is removed from scripting.

Reviewer: Minor nit: `has been removed` is a little more consistent with our other deprecations.

Feel free to ignore if wanted.
```suggestion
.The `JodaCompatibleDateTime` class has been removed.
```

Author: Fixed locally.

[%collapsible]
====
*Details* +
As a transition from Joda datetime to Java datetime, scripting used

> an intermediate class called `JodaCompatibleDateTime`. This class is
> now removed, and is replaced by `ZonedDateTime`.

Reviewer: Same as above. Feel free to ignore.
```suggestion
an intermediate class called `JodaCompatibleDateTime`. This class has
been removed and is replaced by `ZonedDateTime`.
```

Author: Fixed locally.

*Impact* +

> Prior to upgrade, any scripts using `getDayOfWeek` must replace
> this with `getDayOfWeekEnum().value`. After upgrade, any scripts
> using `getDayOfWeekEnum` should replace this method with `getDayOfWeek`
> as it now returns the `DayOfWeek` enum instead of an `int` value.

Reviewer: This is mostly minor nits you can ignore if wanted.

We may also want to explain what will happen if users don't replace these methods in their scripts.
```suggestion
Before upgrading, replace `getDayOfWeek` with `getDayOfWeekEnum().value` in any
scripts.

After upgrading, replace `getDayOfWeekEnum` with `getDayOfWeek` in any scripts.
`getDayOfWeek` now returns the `DayOfWeek` enum instead of an `int` value.
```

Author: Updated.
